src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -53,6 +51,7 @@
   private static final String donutsPackage = "com.squareup.donuts";
 
   @Rule public final CompilationRule compilation = new CompilationRule();
+  @Rule public final ExpectedException expectedException = ExpectedException.none();
```

Comment:
* Can't have multiple methods that fail in a single test, can have multiple try/catch blocks.
* Can't assert things like the `getCause()`, regex on the message, elements in the stack trace.
* Can't assert the exact statement that should throw, any matching exception passes test vs. only a single expression with try/catch